includes/class-wc-swedbank-plugin.php

```diff
@@ -0,0 +1,554 @@
+	 */
+	public function ajax_swedbank_pay_capture() {
+		if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'swedbank_pay' ) ) {
+			exit( 'No naughty business' );
+		}
+
+		$order_id = (int) $_REQUEST['order_id'];
+		$order    = wc_get_order( $order_id );
+
+		// Get Payment Gateway
+		$gateways = WC()->payment_gateways()->get_available_payment_gateways();
+		if ( isset( $gateways[ $order->get_payment_method() ] ) ) {
+			$gateway = $gateways[ $order->get_payment_method() ];
+
+			try {
+				$gateway->capture_payment( $order_id );
+				wp_send_json_success( __( 'Capture success.', 'swedbank-pay-woocommerce-payments' ) );
+			} catch ( Exception $e ) {
+				$message = $e->getMessage();
+				wp_send_json_error( $message );
+			}
+		}
+	}
+
+	/**
+	 * Action for Cancel
+	 */
+	public function ajax_swedbank_pay_cancel() {
+		if ( ! wp_verify_nonce( $_REQUEST['nonce'], 'swedbank_pay' ) ) {
+			exit( 'No naughty business' );
+		}
+
+		$order_id = (int) $_REQUEST['order_id'];
+		$order    = wc_get_order( $order_id );
+
+		// Get Payment Gateway
+		$gateways = WC()->payment_gateways()->get_available_payment_gateways();
+		if ( isset( $gateways[ $order->get_payment_method() ] ) ) {
+			$gateway = $gateways[ $order->get_payment_method() ];
+
+			try {
+				$gateway->cancel_payment( $order_id );
+				wp_send_json_success( __( 'Cancel success.', 'swedbank-pay-woocommerce-payments' ) );
+			} catch ( Exception $e ) {
+				$message = $e->getMessage();
+				wp_send_json_error( $message );
+			}
+		}
+	}
+
+	/**
+	 * Generate UUID
+	 *
+	 * @param $node
+	 *
+	 * @return string
+	 */
+	public function generate_uuid( $node ) {
+		return \Ramsey\Uuid\Uuid::uuid5( \Ramsey\Uuid\Uuid::NAMESPACE_OID, $node )->toString();
+	}
+
+	/**
+	 * Dispatch Background Process
+	 */
+	public function maybe_process_queue() {
+		self::$background_process->dispatch();
+	}
+
+	/**
+	 * Provide Admin Menu items
+	 */
+	public function admin_menu() {
+		// Add Upgrade Page
+		global $_registered_pages;
+
+		$hookname = get_plugin_page_hookname( 'wc-payex-psp-upgrade', '' );
+		if ( ! empty( $hookname ) ) {
+			add_action( $hookname, __CLASS__ . '::upgrade_page' );
+		}
+
+		$_registered_pages[ $hookname ] = true;
+	}
+
+	/**
+	 * Upgrade Page
+	 */
+	public static function upgrade_page() {
+		if ( ! current_user_can( 'update_plugins' ) ) {
+			return;
+		}
+
+		// Run Database Update
+		include_once( dirname( __FILE__ ) . '/class-wc-swedbank-pay-update.php' );
+		\WC_Swedbank_Pay_Update::update();
+
+		echo esc_html__( 'Upgrade finished.', 'swedbank-pay-woocommerce-payments' );
+	}
+
+	/**
+	 * Upgrade Notice
+	 */
+	public static function upgrade_notice() {
+		if ( current_user_can( 'update_plugins' ) ) {
+			?>
+			<div id="message" class="error">
+				<p>
+					<?php
+					echo esc_html__(
+						'Warning! Swedbank Pay WooCommerce payments plugin requires to update the database structure.',
+						'swedbank-pay-woocommerce-payments'
+					);
+					echo ' ' . sprintf(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of echo language construct is discouraged.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-woocommerce-payments/pullRequest?prid=5340032)